content/docs/migration_1.0/C_Pico.md

```diff
@@ -0,0 +1,802 @@
+---
+title: "C / Pico"
+weight : 6400
+menu:
+  docs:
+    parent: migration_1.0
+---
+
+## General API changes
+
+We have reworked the type naming to clarify how types should be interacted with.
+
+### Owned types
+
+Owned types are allocated by the user and it is their responsibility to drop them using `z_drop` (or `z_close` for sessions).
+
+Previously, we were returning Zenoh structures by value. In Zenoh 1.0.0, a reference to memory must be provided. This allows initializing user allocated structures and frees return value for error codes.
+
+Here is a quick example of this change:
+
+- Zenoh 0.11.x
+
+```c
+z_owned_session_t session = z_open(z_move(config));
+if (!z_check(session)) {
+    return -1;
+}
+z_close(z_move(session));
+```
+
+- Zenoh 1.0.0
+
+```c
+z_owned_session_t session;
+if (z_open(&sesion, z_move(config), &opts) < 0) {
+  return -1;
+}
+z_close(z_move(session))
+```
+
+The owned objects have a “null” state.
+
+- If the constructor function (e.g. `z_open`) fails, the owned object will be set to its null state.
+- The `z_drop` releases the resources of the owned object and sets it to the null state to avoid double drop. Calling `z_drop` on an object in a null state is safe and does nothing.
+- Calling z_drop on an uninitialized object is invalid.
+
+(TODO) Owned types support move semantics, which will consume the owned object and turn it into a moved object, see next section.
+
+### Moved types
+
+Moved types are obtained when using `z_move` on an owned type object. They are consumed on use when passed to relevant functions. Any non-constructor function accepting a moved object (i.e. an object passed by owned pointer) becomes responsible for calling drop on it. The object is guaranteed to be in the null state upon such function return, even if it fails.
+
+### Loaned types
+
+Each owned type now has a corresponding `z_loaned_xxx_t` type, which is obtained by calling
+
+ `z_loan` or `z_loan_mut` on it, or eventually received from Zenoh functions / callbacks.
+
+It is no longer possible to directly access the fields of an owned object, the accessor functions on the loaned objects should instead be used.
+
+ Here is a quick example:
+
+- Zenoh 0.11.x
+
+```c
+void reply_handler(z_owned_reply_t *reply, void *ctx) {
+    if (z_reply_is_ok(reply)) {
+        z_sample_t sample = z_reply_ok(reply);
+        printf(">> Received ('%.*s')\n", (int)sample.payload.len, sample.payload.start);
+    }
+}
+```
+
+- Zenoh 1.0.0
+
+```c
+void reply_handler(const z_loaned_reply_t *reply, void *ctx) {
+    if (z_reply_is_ok(reply)) {
+        const z_loaned_sample_t *sample = z_reply_ok(reply);
+        z_owned_string_t replystr;
+        z_bytes_deserialize_into_string(z_sample_payload(sample), &replystr);
+        printf(">> Received ('%s')\n", z_string_data(z_loan(replystr)));
+    }
+}
+```
+
+Certain loaned types can be copied into owned, using the `z_clone` function:
```

Comment:
by checking the docs ? Actually pico implements clone for majority of owned objects, while zenoh-c only for a small subset of them. (Since not every object is clonable in rust).

Reply:
Then let's state it explicitly that to know what are *certain loaned types* documentation should be checked. Is it stated explicitly in the documentation that an object can or can not be copied?

Reply:
It is not stated explicitly, but all objects that can be cloned have a document z_xxx_clone function.

Reply:
Added a comment to consult docs for function signature.